src/goto-symex/symex_goto.cpp

```diff
@@ -210,17 +210,24 @@ void goto_symext::symex_goto(statet &state)
   // merge_gotos when we visit new_state_pc
   framet::goto_state_listt &goto_state_list =
     state.call_stack().top().goto_state_map[new_state_pc];
-  goto_state_list.emplace_back(state);
 
-  symex_transition(state, state_pc, backward);
-
-  // adjust guards
+  // If the guard is unconditionally true it means the alternative branch is
```

Comment:
Not so -- (1) there is no one unique alternative branch, there can be several paths around this code; (2) this isn't the state guard, it's the GOTO condition. `new_guard.is_true()` just means an unconditional jump, regardless of whether there are paths around this code.

Suggested phrasing: // On an unconditional GOTO we don't need our state any more, as it will be overwritten by `merge_goto`. Therefore we move it onto `goto_state_list` instead of copying as usual.